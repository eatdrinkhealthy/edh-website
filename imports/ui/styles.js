import { css } from "glamor";

//
// Component Style Pattern (use with glamor)
//
// const componentRules = {
//    rule1: "value",
//      ...
//    ruleN: "value",
// };
// export const componentStyle = {
//    rules: componentRules,
//    className: `${css(componentRules)}`,
// };
//

const appLinksHeaderRules = {
  marginTop: "90px",
  marginLeft: "40px",
  fontSize: "16px",
  color: "darkbrown",
};
export const appLinksHeaderStyle = {
  rules: appLinksHeaderRules,
  className: `${css(appLinksHeaderRules)}`,
};

const appLinksButtonsRules = {
  marginTop: "14px",
  marginBottom: "20px",
  marginLeft: "40px",
};
export const appLinksButtonsStyle = {
  rules: appLinksButtonsRules,
  className: `${css(appLinksButtonsRules)}`,
};

const linkButtonRules = {
  backgroundColor: "darkslategray",
  border: "none",
  borderRadius: "4px",
  color: "white",
  padding: "8px 32px 8px 32px",
  marginRight: "8px",
  textAlign: "center",
  textDecoration: "none",
  fontSize: "12px",
};
export const linkButtonStyle = {
  rules: linkButtonRules,
  className: `${css(linkButtonRules)}`,
};
