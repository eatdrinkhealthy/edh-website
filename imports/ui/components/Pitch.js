import React from "react";
import { pitchStyle } from "../styles";

const Pitch = () => (
  <div className={`${pitchStyle.className} anotherClass`}>
    A platform to find, share, and discuss healthy places to shop and eat.
  </div>
);

Pitch.style = pitchStyle;

export default Pitch;
