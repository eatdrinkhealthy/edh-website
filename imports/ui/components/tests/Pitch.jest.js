/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
import React from "react";
import renderer from "react-test-renderer";
import Pitch from "../Pitch";
import {
  formattedJSON,
  getAllComponentStyle,
} from "../../../utils/tests/testUtils";
import * as styles from "../../../ui/styles";

describe("<Pitch />", function () {
  it("matches render snapshot", function () {
    const tree = renderer.create(<Pitch />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches style / css snapshot", function () {
    const allStyle = getAllComponentStyle(<Pitch />, styles);
    expect(formattedJSON(allStyle)).toMatchSnapshot();
  });
});
