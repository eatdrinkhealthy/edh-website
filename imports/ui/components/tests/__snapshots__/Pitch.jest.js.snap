exports[`<Pitch /> matches render snapshot 1`] = `
<div
  className="css-1dl4n80">
  A platform to find, share, and discuss healthy places to shop and eat.
</div>
`;

exports[`<Pitch /> matches style / css snapshot 1`] = `
"[
  {
    \"rules\": {
      \"paddingTop\": \"30px\",
      \"fontSize\": \"15px\",
      \"color\": \"white\",
      \"display\": \"block\",
      \"width\": \"226px\",
      \"margin\": \"0 auto\",
      \"paddingBottom\": \"30px\",
      \"borderBottom\": \"1px solid rgba(255,255,255,0.6)\"
    },
    \"className\": \"css-1dl4n80\"
  }
]"
`;
